backend/community_xp/services.py

```
@@ -467,6 +448,28 @@ def _apply_sync_run_unlocked(run, applied_by=None):
+                status=Mee6SyncRun.STATUS_SUCCESS,
+                completed_at__isnull=False,
+                applied_at__isnull=False,
+            )
+            .exclude(id=run.id)
+            .order_by('-completed_at', '-id')
+            .first()
+        )
+        if latest_applied_run and run.completed_at < latest_applied_run.completed_at:
+            raise Mee6SyncError(
+                f'Cannot apply MEE6 sync #{run.id}; sync #{latest_applied_run.id} is newer and already applied'
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)